### PR TITLE
30 august chat room

### PR DESCRIPTION
**Pull request by ShahirMansuri786**

### CODE REVIEW

File: app/controllers/messages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    @message = Message.new
+  end
+
+  # GET /messages/1/edit
+  def edit
+  end
+
+  # POST /messages or /messages.json
+  def create
+    @message = Message.new(message_params)
```
Comment (aadi163) [inline app/controllers/messages_controller.rb]: why did you use instance variable?

File: app/controllers/messages_controller.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+       redirect_to request.referrer
+      else
+        
+      end
+    
+  end
+
+  # PATCH/PUT /messages/1 or /messages/1.json
+  def update
+
```
Comment (aadi163) [inline app/controllers/messages_controller.rb]: why are you writing code in next line inside the action ?

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -7,7 +7,7 @@ def user_edit
     @user = User.find(params[:id])
   end
 
-  def show
+  def show_user
      @user = User.find(params[:id])
```
Comment (aadi163) [inline app/controllers/users_controller.rb:11]: you can use local variable.

File: app/controllers/rooms_controller.rb [hunk 4]
```diff
@@ -1,62 +1,45 @@
...
-      end
-  
+    if @room.update(room_params)
+    else 
+    end
   end
 
-  # DELETE /rooms/1 or /rooms/1.json
   def destroy
+    byebug
```
Comment (aadi163) [inline app/controllers/rooms_controller.rb]: you need to remove byebug.

File: app/controllers/rooms_controller.rb [hunk 5]
```diff
@@ -1,62 +1,45 @@
...
   def update
-    
-      if @room.update(room_params)
-       
-      else
-       
-      end
-  
+    if @room.update(room_params)
+    else
```
Comment (aadi163) [inline app/controllers/rooms_controller.rb:29]: Why are you using unnecessary if else.

File: app/controllers/rooms_controller.rb [hunk 6]
```diff
@@ -1,62 +1,45 @@
...
     @room = Room.new(room_params)
-
-   
-      if @room.save
-      
-      else
-        
-      end
-  
+    if @room.save
```
Comment (aadi163) [inline app/controllers/rooms_controller.rb]: Here you can use local variable as well.

File: app/controllers/messages_controller.rb (hunk 1 above)
Comment (ShahirMansuri786) [reply app/controllers/messages_controller.rb]: ok i will update it

File: app/controllers/messages_controller.rb (hunk 2 above)
Comment (ShahirMansuri786) [reply app/controllers/messages_controller.rb]: I have changed all the things in another commit

File: app/controllers/rooms_controller.rb (hunk 4 above)
Comment (ShahirMansuri786) [reply app/controllers/rooms_controller.rb]: I have removed in another commit

File: app/controllers/rooms_controller.rb (hunk 5 above)
Comment (ShahirMansuri786) [reply app/controllers/rooms_controller.rb:29]: currently i am working on this feature so for future requirement i have wrote this if no needed so i will update it

File: app/controllers/rooms_controller.rb (hunk 6 above)
Comment (ShahirMansuri786) [reply app/controllers/rooms_controller.rb]: sure i wiil update

File: app/controllers/rooms_controller.rb (hunk 5 above)
Comment (vinayaklovanshi) [reply app/controllers/rooms_controller.rb:29]: Remove that else, whenever you will need it you can add later.

File: app/controllers/rooms_controller.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    end
+  end
+
+  def destroy
+    if @room.destroy!
+    end
+  end
+
+  private
+    def set_room
```
Comment (vinayaklovanshi) [inline app/controllers/rooms_controller.rb:39]: correct the intent

File: app/controllers/rooms_controller.rb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def update
+    if @room.update(room_params)
+    else 
+    end
+  end
+
+  def destroy
+    if @room.destroy!
```
Comment (vinayaklovanshi) [inline app/controllers/rooms_controller.rb:34]: Why you added if condition here ?

File: app/controllers/rooms_controller.rb [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        @room = Room.find(params[:id])
+      end  
+    end
+
+    def room_params
+      params.require(:room).permit(:name)
+    end
+
+    
+end
```
Comment (vinayaklovanshi) [inline app/controllers/rooms_controller.rb:50]: Why extra space here ?

File: app/controllers/rooms_controller.rb (hunk 5 above)
Comment (ShahirMansuri786) [reply app/controllers/rooms_controller.rb:29]: ok sir i will update it

File: app/controllers/rooms_controller.rb (hunk 7 above)
Comment (ShahirMansuri786) [reply app/controllers/rooms_controller.rb:39]: sure sir

File: app/controllers/rooms_controller.rb (hunk 8 above)
Comment (ShahirMansuri786) [reply app/controllers/rooms_controller.rb:34]: ok sir i will correct it

File: app/controllers/rooms_controller.rb (hunk 9 above)
Comment (ShahirMansuri786) [reply app/controllers/rooms_controller.rb:50]: I will correct sir

**Pull request merged**